src/rules/no-object-math/rule.spec.ts

```diff
@@ -8,26 +8,273 @@ const messageId = "object-math-violation";
+	`,
+	unindent`
+		const position = new Vector3(0, 0, 0);
+		const scaled = position.mul(2);
+	`,
+	unindent`
+		const cf1 = new CFrame();
+		const cf2 = new CFrame();
+		const result = cf1.mul(cf2);
+	`,
+	unindent`
+		const v1 = new Vector2(1, 2);
+		const v2 = new Vector2(3, 4);
+		const v3 = new Vector2(5, 6);
+		const result = v1.add(v2).mul(v3);
+	`,
+	unindent`
+		const pos = new Vector3(1, 2, 3);
+		const chained = pos.mul(2).add(new Vector3(1, 1, 1)).div(3);
+	`,
 	"const v3 = new Vector3(1, 2, 3); const v4 = new Vector3(4, 5, 6); const sum = v3.add(v4);",
 	"const cf1 = new CFrame(); const cf2 = new CFrame(); const product = cf1.mul(cf2);",
 	"const u1 = new UDim2(); const u2 = new UDim2(); const diff = u1.sub(u2);",
 	"const vec2 = new Vector2(10, 10); const quotient = vec2.div(2);",
 ];
 
 const invalid: Array<InvalidTestCase> = [
-	// Invalid math operators
+	{
+		code: unindent`
+			const vector1 = new Vector2(1, 2);
+			const vector2 = new Vector2(3, 4);
+			const result = vector1 + vector2;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const vector1 = new Vector2(1, 2);
+				const vector2 = new Vector2(3, 4);
+				const result = vector1.add(vector2);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const vector = new Vector2(1, 2);
+			const scaled = vector * 2;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const vector = new Vector2(1, 2);
+				const scaled = vector.mul(2);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const pos1 = new Vector3(1, 2, 3);
+			const pos2 = new Vector3(4, 5, 6);
+			const combined = pos1 - pos2;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const pos1 = new Vector3(1, 2, 3);
+				const pos2 = new Vector3(4, 5, 6);
+				const combined = pos1.sub(pos2);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const position = new Vector3(1, 2, 3);
+			const divided = position / 2;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const position = new Vector3(1, 2, 3);
+				const divided = position.div(2);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const cf1 = new CFrame();
+			const cf2 = new CFrame();
+			const result = cf1 * cf2;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const cf1 = new CFrame();
+				const cf2 = new CFrame();
+				const result = cf1.mul(cf2);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const v1 = new Vector2(1, 2);
+			const v2 = new Vector2(3, 4);
+			const v3 = new Vector2(5, 6);
+			const result = v1 + v2 + v3;
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const v1 = new Vector2(1, 2);
+				const v2 = new Vector2(3, 4);
+				const v3 = new Vector2(5, 6);
+				const result = v3.add(v1.add(v2));"
```

Comment:
I'm not sure this matters, but this autofix results in slightly different emit:
```luau
local _arg0 = v1 + v2
local result = v3 + _arg0
```
`v3 +` rather than `+ v3`